sql/catalyst/src/main/scala-2.12/org/apache/spark/sql/catalyst/expressions/AttributeMap.scala

```diff
@@ -23,6 +23,10 @@ package org.apache.spark.sql.catalyst.expressions
  * of the name, or the expected nullability).
  */
 object AttributeMap {
+  def apply[A](kvs: Map[Attribute, A]): AttributeMap[A] = {
```

Comment:
instead of adding a new API, can the caller side just do `AttributeMap(map.toSeq)`?

Reply:
That would work as well, but `AttributeMap(kvs = map.toSeq)` will end up calling `.toMap` down the line.

Reply:
Ah got it, this is more efficient